### DISCUSSION TITLE
0  my cypress script run for react.js base appliction in that click on "button" & open pop window but when click on button popup not open.but when click manually this time it will be open.how to ressolve that issue.

### DISCUSSION THREAD

**Discussion by Willingdon1113** [cypress-io/cypress#28053] [Questions and Help]
This is my cypress.config.js file

const { defineConfig } = require("cypress");

module.exports = defineConfig({ e2e: { baseUrl: "http://127.0.0.1:8000/dashboard", projectId: "o4k5hc", setupNodeEvents(on, config) { // implement node event listeners here }, testIsolation: false,

}, includeShadowDom: true,

component: { devServer: { framework: "react", bundler: "vite", }, }, });

"includeShadowDom: true," this method also use for the shadow dom then will be it not worked.

**Comment by portyu9** [comment]
Hi @Willingdon1113 

`includeShadowDom: true` will not solve this. That option only allows Cypress queries such as `cy.get()` to find elements inside Shadow DOM; it has no effect on browser popup windows or new tabs.

First check how the button opens the popup.

If it calls `window.open()`, verify that call directly:

```js id="ze71d2"
cy.window().then((win) => {
  cy.stub(win, 'open').as('windowOpen')
})

cy.get('button').click()

cy.get('@windowOpen').should('have.been.called')
```

You can also assert the URL:

```js id="ogjjml"
cy.get('@windowOpen')
  .should('have.been.calledWithMatch', /expected-url/)
```

If the call is made, the button is working—the remaining issue is the new browser window, not the click itself.

If the element is a link with `target="_blank"`, a common Cypress approach is to keep the navigation in the same tab:

```js id="9ttnrb"
cy.get('a')
  .invoke('removeAttr', 'target')
  .click()
```

and then assert the destination page.

Cypress traditionally runs the test in a single browser tab, so I would not expect `includeShadowDom`, `testIsolation: false`, or additional waits to make a real popup controllable.

If your test specifically requires interacting with a real second browser window/tab, current Cypress supports that through the official `@cypress/puppeteer` integration.

Also, if the button itself is inside a shadow root, then `includeShadowDom: true` can help Cypress locate the button—but it still does not control the popup that the button opens.
